core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala

```diff
@@ -792,10 +801,23 @@ class SmartConnectorRowRDD(_session: SnappySession,
   }
 
   override def computeResultSet(
-      thePart: Partition): (Connection, Statement, ResultSet) = {
+      thePart: Partition, context: TaskContext): (Connection, Statement, ResultSet) = {
     val helper = new SparkShellRDDHelper
     val conn: Connection = helper.getConnection(
       connProperties, thePart)
+    if (context ne null) {
+      val partitionId = context.partitionId()
+      context.addTaskCompletionListener { _ =>
+        logDebug(s"closed connection for task from listener $partitionId")
+        try {
+          conn.commit()
+          conn.close()
+          logDebug("closed connection for task " + context.partitionId())
+        } catch {
+          case NonFatal(e) => logWarning("Exception closing connection", e)
+        }
+      }
+    }
```

Comment:
@sumwale ColumnInsertExec doesn't depend on TaskContext. We get a PoolConnection/EmbedConnection and close it ourselves. Similarly for tx commit/rollback.

To iterate over rowbuffer, we use embed connection and don't do  any commit/rollback. It just inherits the txState of thread which was doing put into rowbuffer.

This change looks to be safe.